### PR TITLE
ldap/auth: add tests for login regressions

### PR DESCRIPTION
**Pull request by fairclothjm**
Add tests to cover the regressions reported in
- https://github.com/hashicorp/vault/issues/26171
- https://github.com/hashicorp/vault/issues/26183

### CODE REVIEW

File: builtin/credential/ldap/backend_test.go [hunk 1]
```diff
@@ -418,9 +418,79 @@ func factory(t *testing.T) logical.Backend {
...
+			// Authenticate
+			testAccStepLogin(t, "hermes", "hermes"),
+
+			// Verify both groups mappings can be listed back
+			testAccStepGroupList(t, []string{"engineers", "admin_staff"}),
+
+			// Verify user mapping can be listed back
+			testAccStepUserList(t, []string{"hermes"}),
+		},
+	})
```
Comment (thyton) [inline builtin/credential/ldap/backend_test.go:488]: Looks like `TestBackend_LoginRegression_UserAttr`'s content is the same as `TestBackend_LoginRegression_AnonBind`'s L424-452. I wonder if it's intended to lay out future work.

Comment (fairclothjm) [reply builtin/credential/ldap/backend_test.go:488]: oh, woops looks like I accidentally added `cfg.AnonymousGroupSearch = true` here. That is the only difference between the setup of the two regression cases. I will remove that from this test

Comment (thyton) [reply builtin/credential/ldap/backend_test.go:488]: Got it. Thanks!

**Review by thyton — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**